src/cpp/r/session/graphics/RGraphicsDevDesc.hpp

```diff
@@ -55,6 +55,7 @@ struct DevDescVersion9
    Rboolean canGenMouseMove;
    Rboolean canGenMouseUp;
    Rboolean canGenKeybd;
+   Rboolean canGenIdle; // version 12
```

Comment:
I'm setting it here now: https://github.com/rstudio/rstudio/pull/1090/files#diff-dcfd8bfc2c362a35a03b7913472834d8R734

Note that I moved `setDeviceAttributes()` to be a templated function with dispatch into `RGraphicsDevDesc.cpp`, just to make sure we don't run afoul of any layout differences.